### PR TITLE
Release/2.0.0

### PR DESCRIPTION
**Pull request by aboedo**
## 2.0.0

- removes deprecated `getEntitlements`, replaced by `getOfferings`
- removes deprecated `makePurchase`, replaced by `purchaseProduct`

- iOS: 
    - added new method, `syncPurchases`, that enables syncing the purchases in the local receipt with the backend without risking a password prompt. The method was already available on Android.
    - added a new method, `presentCodeRedemptionSheet`, for offer codes redemption.
- Bumped purchases-hybrid-common to 2.0.0 [Changelog here](https://github.com/RevenueCat/purchases-hybrid-common/releases/2.0.0)
- Bumped purchases-ios to 3.9.2 [Changelog here](https://github.com/RevenueCat/purchases-ios/releases/3.9.1)
- Bumped purchases-android to 4.0.0 [Changelog here](https://github.com/RevenueCat/purchases-hybrid-common/releases/4.0.0)

### CODE REVIEW

**Comment by aboedo** [comment]
tests are going to keep failing until purchasesHybridCommon is released

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,15 @@
...
+
+- removes deprecated `getEntitlements`, replaced by `getOfferings`
+- removes deprecated `makePurchase`, replaced by `purchaseProduct`
+
+- iOS: 
+    - added new method, `syncPurchases`, that enables syncing the purchases in the local receipt with the backend without risking a password prompt. The method was already available on Android.
+    - added a new method, `presentCodeRedemptionSheet`, for offer codes redemption.
+- Bumped purchases-hybrid-common to 1.5.0 [Changelog here](https://github.com/RevenueCat/purchases-hybrid-common/releases/1.5.0)
+- Bumped purchases-ios to 3.9.2 [Changelog here](https://github.com/RevenueCat/purchases-ios/releases/3.9.1)
+- Bumped purchases-android to 4.0.0 [Changelog here](https://github.com/RevenueCat/purchases-hybrid-common/releases/4.0.0)
```
Comment (vegaro) [inline CHANGELOG.md:11]: This should point to 4.0.1

File: example/android/build.gradle [hunk 2]
```diff
@@ -5,7 +5,7 @@ buildscript {
     }
 
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.5.0'
+        classpath 'com.android.tools.build:gradle:4.0.1'
```
Comment (vegaro) [inline example/android/build.gradle:8]: This update is the reason why CircleCI is failing

File: example/android/gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -1,6 +1,6 @@
-#Mon Jun 15 14:41:47 PDT 2020
+#Wed Dec 16 14:24:29 UYT 2020
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.1.1-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.5-all.zip
```
Comment (vegaro) [inline example/android/gradle/wrapper/gradle-wrapper.properties:6]: I think you might need to revert this change too when reverting the gradle android plugin to 3.5.0, but I am not sure

File: example/android/build.gradle (hunk 2 above)
Comment (aboedo) [reply example/android/build.gradle:8]: interesting. Android studio nagged me about it so I just followed along, I'll revert

**Review by vegaro — APPROVED** [review]
